lib/yext/api/utils/middleware/response_parser.rb

```diff
@@ -46,36 +47,62 @@ def parse(body)
             #
             #   { title: [{ error: 'blank'}, { error: 'too_short', count: 10 }]}
 
-            # yext_response_json = MultiJson.load(body, symbolize_keys: true)
-            return {} if body.starts_with?("<html")
-            return { data: [{ response_string: body }], metadata: {}, errors: {} } unless body.starts_with?("{")
+            if body.starts_with?("<html")
+              save_meta_data(parse_env, {}, response_env[:status], body: body)
 
-            yext_response_json = JSON.parse(body, symbolize_names: true)
+              return {}
+            end
+
+            @yext_response_json = {}
 
-            data = extract_data_value(yext_response_json)
+            data      = data_value(body)
+            meta_data = meta_data_value
 
-            save_meta_data(Yext::Api::Utils::ApiFinder.new(parse_env[:url], parse_env[:method]).find_api, yext_response_json[:meta])
+            save_meta_data(parse_env, meta_data, response_env[:status], data)
 
-            { data:     data,
-              metadata: yext_response_json[:meta],
-              errors:   {} }
+            { data: data, metadata: meta_data, errors: {} }
```

Comment:
Is it safe to assume there are no errors here?

Reply:
Absolutely not.

The problem is that I don't know what errors in Yext will look like nor if I will be able to parse them out into the format that Spyke wants them.

Based on what I've seen so far the second point is far more valid.

At this time, I'm leaving the errors in meta and figuring that the consumer will have to deal with the errors.

It isn't a good solution, but because this is middleware, I have to deal with errors fairly generically, and I don't have a good generic solution/idea for this at this time.

This is probably something we'll fix in later versions of the gem, but I had to start somewhere.